core/matching/side.go

```diff
@@ -160,7 +160,7 @@ func (s *OrderBookSide) amendOrder(orderAmend *types.Order) (uint64, error) {
 		return 0, types.ErrOrderAmendFailure
 	}
 
-	if oldOrder.Size != orderAmend.Size &&
+	if oldOrder.Size < orderAmend.Size &&
 		oldOrder.Remaining < orderAmend.Size {
```

Comment:
Does this line need to be here? I think if line 163 is true we are in trouble because we cannot update in place.

Reply:
yea indeed, that's the point. If order the newOrder have an increased side, that's a bug. and should never happend.